src/components/BooksForm.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { v4 as uuidv4 } from 'uuid';
+import categories from './categories';
+import { createBook } from '../actions/index';
+
+class BooksForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: 'Action',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.resetForm = this.resetForm.bind(this);
+  }
+
+  handleChange(e) {
+    const { title, category } = this.state;
+    const { value, nodeName } = e.target;
+
+    let newTitle = title;
+    let newCategory = category;
+    switch (nodeName) {
+      case 'INPUT':
+        newTitle = value;
+        break;
+      case 'SELECT':
+        newCategory = value;
+        break;
+      default:
+        newTitle = title;
+    }
+
+    this.setState({ category: newCategory, title: newTitle });
+  }
```

Comment:
this can also be done in short form as
```
handleChange(event) {
  const name = event.target.name;

  this.setState({
    title: event.target.value
    [name]: event.target.value
  })
}
```